### DISCUSSION TITLE
how do I add the arabic keyboard layout

### DISCUSSION THREAD

**Discussion by pixel-anas** [i3/i3#4767] [Q&A]
I want to add an arabic keyboard layout to i3wm and put alt + shift as a shortcut to switch between english and arabic

**Comment by Airblader** [comment]
Which keyboard layouts are supported or which one you have chosen has nothing to do with i3 as the window manager – we simply "consume" the layout you have chosen using e.g. `setxkbmap`. You can refer to resources like [this](https://wiki.archlinux.org/title/Xorg/Keyboard_configuration) for documentation on choosing layouts.

**Comment by mohamadali-halwani** [comment]
You add this line to your i3 config file located in `~/.config/i3/`
```
# Keyboard Layouts
exec setxkbmap -model pc105 -layout us,ar -option 'grp:alt_shift_toggle'
```

**Reply by AliLtRP** [reply]
> You add this line to your i3 config file located in `~/.config/i3/`
> 
> ```
> # Keyboard Layouts
> exec setxkbmap -model pc105 -layout us,ar -option 'grp:alt_shift_toggle'
> ```

finally :)

**Comment by osama-mhmd** [comment]
also you should make sure that you have the arabic language installed by running:
```sh
localectl list-x11-keymap-layouts | grep ar
```

`You may find it 'ara'`, so you have to change the command to `exec setxkbmap -model pc105 -layout us,ara -option 'grp:alt_shift_toggle'`

**Comment by Mahmoud12344** [comment]
exec_always --no-startup-id setxkbmap -layout us, ara -option grp:alt_shift_toggle

change the `ar` r to `ara` as ar has been *deprecated* at the time of writing this
